syntax = "proto3";
package repository;

import "google/protobuf/timestamp.proto";
import "common.proto";

option go_package = "github.com/EdmilsonRodrigues/ophelia-ci";

service RepositoryService {
    rpc CreateRepository(CreateRepositoryRequest) returns (RepositoryResponse);
    rpc UpdateRepository(UpdateRepositoryRequest) returns (RepositoryResponse);
    rpc ListRepository(common.Empty) returns (ListRepositoryResponse);
    rpc GetRepository(GetRepositoryRequest) returns (RepositoryResponse);
    rpc DeleteRepository(DeleteRepositoryRequest) returns (common.Empty);
}

message GetRepositoryRequest {
    string id = 1;
    string name = 2;
}

message CreateRepositoryRequest {
    string name = 1;
    string description = 2;
    string gitignore = 3;
}

message UpdateRepositoryRequest {
    string id = 1;
    string name = 2;
    string description = 3;
}

message DeleteRepositoryRequest {
    string id = 1;
}

message RepositoryResponse {
    string id = 1;
    string name = 2;
    string description = 3;
    google.protobuf.Timestamp last_update = 4;
}

message ListRepositoryResponse {
    repeated RepositoryResponse repositories = 1;
}
